src/Bicep.Core.Samples/Files/InvalidFunctions_LF/main.bicep

```diff
@@ -0,0 +1,35 @@
+func useRuntimeFunction() string => reference('foo').bar
+
+func constFunc() string => 'A'
+func funcWithOtherFuncRef() string => constFunc()
+
+func missingArgType(input) string => input
+
+func missingOutputType(input string) => input
+
+func invalidType(input string) string => input
+
+output invalidType string = invalidType(true)
+
+func madeUpTypeArgs(a notAType, b alsoNotAType) string => '${a}-${b}'
+
+func noLambda('foo') string => ''
+
+func noLambda2 = (sdf 'foo') string => ''
+
+func noLambda3 = string 'asdf'
+
+func argLengthMismatch(a string, b string, c string) array => ([a, b, c])
+var sdf = argLengthMismatch('asdf')
+
+var asdfwdf = noLambda('asd')
+
+func sayHello(name string) string => 'Hi ${name}!'
+output hellos array = map(['Evie', 'Casper'], sayHello) // this syntax not supported currently, but should it be?
```

Comment:
Either this, or we'd have to come up with a special syntax for passing a function reference as an argument (e.g. `function('blah')`. This feels like a good option if we want to avoid syntax changes